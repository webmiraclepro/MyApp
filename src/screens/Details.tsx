import React from 'react'
import { StyleSheet, View, Text, Button, Image } from 'react-native'
import { NavigationInjectedProps, SafeAreaView } from 'react-navigation'
import { Transition } from 'react-navigation-fluid-transitions'

import { User } from '../models'

interface Data extends User {
  image: string
}

const DetailsScreen: React.SFC<NavigationInjectedProps> = ({ navigation }) => {

  const { email, image } = navigation.getParam('data', {}) as Data

  return (
    <SafeAreaView forceInset={{top: 'always'}} style={[StyleSheet.absoluteFill, styles.container]}>
      <View style={styles.header}>
        <Button
          onPress={() => navigation.goBack()}
          title="Back"
          color="#841584"
          accessibilityLabel="Go back"
        />
        <Text>Usuario</Text>
      </View>
      <View style={styles.content}>
        <Transition appear='top' shared={email}>
          <View style={styles.card}>
            <Image style={{ flex: 1 }} resizeMode='contain' source={{ uri: image }} />
          </View>
        </Transition>
      </View>
    </SafeAreaView>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#231E32'
  },
  header: {
    backgroundColor: 'green',
    flexDirection: 'row',
    alignItems: 'center',
    alignSelf: 'flex-start',
    height: 50,
    width: '100%'
  },
  content: {
    flex: 1,
    backgroundColor: 'white'
  },
  card: {
    flexDirection: 'column',
    flex: 1,
    justifyContent: 'center',
    backgroundColor: 'white'
  }
})

export default DetailsScreen
