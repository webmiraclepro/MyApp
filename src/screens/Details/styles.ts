import { StyleSheet } from 'react-native'

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#231E32',
  },
  header: {
    backgroundColor: 'green',
    flexDirection: 'row',
    alignItems: 'center',
    alignSelf: 'flex-start',
    height: 50,
    width: '100%',
  },
  content: {
    flex: 1,
    backgroundColor: 'white',
    padding: 20,
  },
  card: {
    marginTop: 20,
    marginBottom: 20,
    flexDirection: 'column',
    flex: 1,
    justifyContent: 'center',
    backgroundColor: 'white',
  },
  photo: {
    height: 260,
  },
  imageGalleryFooter: {
    padding: 10,
    height: 100,
    flexDirection: 'row'
  },
  imageGalleryButtonContainer: {
    borderWidth: 2,
    borderColor: 'black',
    margin: 5,
    padding: 5
  },
  imageGalleryButton: {
    padding: 10,
    backgroundColor: 'gray'
  }
})
