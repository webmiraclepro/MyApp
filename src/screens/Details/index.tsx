import React, { useState, useEffect } from 'react'
import {
  View,
  ScrollView,
  Text,
  Image,
  Linking,
  Button,
  TouchableOpacity
} from 'react-native'
import { NavigationInjectedProps, NavigationScreenProp } from 'react-navigation'
import { SharedElement } from 'react-navigation-shared-element'
import Icon from 'react-native-vector-icons/MaterialIcons'
import { Avatar } from 'react-native-elements'
import range from 'lodash/range'
import R from 'res'

import styles from './styles'
import { Photo } from '../../models'
import { ParallaxHeader } from '../../containers'
import { BrowserService } from '../../services'
import { ImageGallery } from '../../components'

type GalleryImage = {
  id: string,
  src: string
}

const DetailsScreen = ({ navigation }: NavigationInjectedProps) => {
  const [enableGallery, setEnableGallery] = useState(false)
  const [images, setImages] = useState([] as Array<GalleryImage>)
  const photoInfo = navigation.getParam('data') as Photo
  const { id, email, photo, first, last } = photoInfo
  const openUrl = () => photo && BrowserService.openUrl(photo)

  const loadImages = () => {
    const baseIndex = Math.floor(Math.random() * 100)
    const images = [...new Array(5).keys()].map((i) => {
      return {
        id: `image_${i}`,
        src: `https://picsum.photos/id/${i + baseIndex}/400/400`
      }
    })
    setImages(images)
  }

  useEffect(() => {
    loadImages()
  }, [])

  return (
    <>
      <View style={styles.container}>
        <ParallaxHeader
          headerMaxHeight={150}
          renderLeft={() => (
            <Icon.Button
              name="close"
              activeOpacity={0.7}
              onPress={() => navigation.goBack()}
              backgroundColor="transparent"
              underlayColor="transparent"
            />
          )}
          title={`${last}, ${first}`}
          renderRight={() => (
            <Icon.Button
              name="share"
              activeOpacity={0.7}
              onPress={() => Linking.openURL(`mailto:${email}`)}
              backgroundColor="transparent"
              underlayColor="transparent"
            />
          )}
          renderContent={() => (
            <View style={styles.content}>
              <SharedElement id={id}>
                <TouchableOpacity onPress={() => setEnableGallery(true)}>
                  <View style={styles.card}>
                    <Image
                      style={styles.photo}
                      resizeMode="contain"
                      source={{ uri: photo }}
                    />
                  </View>
                </TouchableOpacity>
              </SharedElement>
              <Button onPress={openUrl} title="Open Url" />
              <Text>{R.strings.details.LOREM_IPSUM}</Text>
            </View>
          )}
        />
      </View>
      <ImageGallery
        images={images}
        isVisible={enableGallery}
        onClosePress={() => setEnableGallery(false)}
        renderFooter={() => (
          <ScrollView
            horizontal
            contentContainerStyle={styles.imageGalleryFooter}
          >
            {range(0, 10).map(() => (
              <Avatar
                onPress={loadImages}
                rounded
                icon={{ name: 'home', color: '#000' }}
                size="large"
                activeOpacity={0.7}
                containerStyle={styles.imageGalleryButtonContainer}
                overlayContainerStyle={styles.imageGalleryButton}
              />
            ))}
          </ScrollView>
        )}
      />
    </>
  )
}

DetailsScreen.navigationOptions = { headerShown: false }

// Add the `sharedElements` function to the component, which
// should return a list of shared-elements to transition.
// The `sharedElements` function is called whenever you navigate
// to or from this screen. You can use the provided navigation
// states or trigger or disable animations.
DetailsScreen.sharedElements = (navigation: NavigationScreenProp<any>) => {
  const { id } = navigation.getParam('data', {}) as Photo
  return [id]
}

export default DetailsScreen
