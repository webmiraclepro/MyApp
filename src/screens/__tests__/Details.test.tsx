import 'react-native'
import React from 'react'
import { mount, shallow } from 'enzyme'

import Details from '../Details'

jest.mock('res', () => ({
  strings: {
    details: {
      LOREM_IPSUM: '',
    },
  },
}))

describe('<Details />', () => {
  const mockData = {
    image: 'Image',
  }

  const defaultProps: any = {
    navigation: {
      goBack: jest.fn(),
      getParam: jest.fn(() => ({})),
    },
    data: mockData,
  }

  it('should render', () => {
    const rendered = mount(<Details {...defaultProps} />)
    expect(rendered).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(shallow(<Details {...defaultProps} />)).toMatchSnapshot()
  })
})
