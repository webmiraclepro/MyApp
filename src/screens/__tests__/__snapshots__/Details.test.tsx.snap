// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> renders correctly 1`] = `
<Fragment>
  <View
    style={
      Object {
        "backgroundColor": "#231E32",
        "flex": 1,
      }
    }
  >
    <ParallaxHeader
      headerMaxHeight={150}
      renderContent={[Function]}
      renderLeft={[Function]}
      renderRight={[Function]}
      title="undefined, undefined"
    />
  </View>
  <ImageGallery
    images={Array []}
    isVisible={false}
    onClosePress={[Function]}
    renderFooter={[Function]}
  />
</Fragment>
`;
